sbi/utils/get_nn_models.py

```diff
@@ -229,6 +228,31 @@ def posterior_nn(
         )
     )
 
+    def build_fn_snpe_a(batch_theta, batch_x, num_components):
+        # Extract the number of components from the kwargs, such that
```

Comment:
add short doc why we need this function that is different from the default build fun?